code/modules/telesci/quantum_pad.dm

```diff
@@ -0,0 +1,189 @@
+
+/obj/machinery/power/quantumpad/RefreshParts()
+	var/E = 0
+	for(var/obj/item/weapon/stock_parts/manipulator/M in component_parts)
+		E += M.rating
+	power_efficiency = E
+	
+	E = 0
+	for(var/obj/item/weapon/stock_parts/capacitor/C in component_parts)
+		E += C.rating
+	
+	teleport_speed = initial(teleport_speed)
+	teleport_speed -= (E*10)
+	teleport_cooldown = initial(teleport_cooldown)
+	teleport_cooldown -= (E * 100)
+
+/obj/machinery/power/quantumpad/attackby(obj/item/I, mob/user, params)
+	if(default_deconstruction_screwdriver(user, I))
+		return
+
+	if(istype(I, /obj/item/device/multitool))
+		if(panel_open)
+			var/obj/item/device/multitool/M = I
+			M.buffer = src
+			to_chat(user, "<span class='notice'>You save the data in [I]'s buffer.</span>")
+			return 1
+		else
+			var/obj/item/device/multitool/M = I
+			if(istype(M.buffer, /obj/machinery/power/quantumpad))
+				linked_pad = M.buffer
+				to_chat(user, "<span class='notice'>You link [src] to the one in [I]'s buffer.</span>")
+				return 1
+
+	if(default_part_replacement(user, I))
+		return
+
+	if(default_deconstruction_crowbar(I))
+		return
+
+	return ..()
+
+/obj/machinery/power/quantumpad/update_icon()
+	. = ..()
+
+	if(inoperable() || panel_open)
+		icon_state = "pad-idle-o"
+	else
+		icon_state = initial(icon_state)
+
+/obj/machinery/power/quantumpad/attack_hand(mob/user)
+	. = ..()
+	if(.)
+		return
+	if(panel_open)
+		to_chat(user, "<span class='warning'>The panel must be closed before operating this machine!</span>")
+		return
+	
+	if(istype(get_area(src), /area/shuttle))
+		to_chat(user, "<span class='warning'>This is too unstable a platform for \the [src] to operate on!</span>")
+		return
+
+	if(!powernet)
+		to_chat(user, "<span class='warning'>[src] is not attached to a powernet!</span>")
+		return
+
+	if(!linked_pad || QDELETED(linked_pad))
+		if(!map_pad_link_id || !initMappedLink())
+			to_chat(user, "<span class='warning'>There is no linked pad!</span>")
+			return
+
+	if(world.time < last_teleport + teleport_cooldown)
+		to_chat(user, "<span class='warning'>[src] is recharging power. Please wait [round((last_teleport + teleport_cooldown - world.time)/10)] seconds.</span>")
+		return
+
+	if(teleporting)
+		to_chat(user, "<span class='warning'>[src] is charging up. Please wait.</span>")
+		return
+
+	if(linked_pad.teleporting)
+		to_chat(user, "<span class='warning'>Linked pad is busy. Please wait.</span>")
+		return
+
+	if(linked_pad.inoperable())
+		to_chat(user, "<span class='warning'>Linked pad is not responding to ping.</span>")
+		return
+	src.add_fingerprint(user)
+	doteleport(user)
+
+/obj/machinery/power/quantumpad/proc/sparks()
+	var/datum/effect/effect/system/spark_spread/sparks = new /datum/effect/effect/system/spark_spread()
+	sparks.set_up(5, 1, get_turf(src))
+	sparks.start()
+
+/obj/machinery/power/quantumpad/attack_ghost(mob/observer/dead/ghost)
+	. = ..()
+	if(.)
+		return
+	if(!linked_pad && map_pad_link_id)
+		initMappedLink()
+	if(linked_pad)
```

Comment:
To prevent ghosts going to nullspace, use QDELETED()

Reply:
Fixed